atlasdb-backup/src/test/java/com/palantir/atlasdb/backup/CoordinationServiceUtilitiesTest.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.backup;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableRangeMap;
+import com.google.common.collect.Range;
+import com.google.common.collect.RangeMap;
+import com.palantir.atlasdb.coordination.ValueAndBound;
+import com.palantir.atlasdb.internalschema.InternalSchemaMetadata;
+import com.palantir.atlasdb.internalschema.InternalSchemaMetadataState;
+import com.palantir.atlasdb.internalschema.TimestampPartitioningMap;
+import com.palantir.timestamp.FullyBoundedTimestampRange;
+import java.util.Map;
+import java.util.Optional;
+import org.junit.Test;
+
+public class CoordinationServiceUtilitiesTest {
+    @Test
+    public void correctlyBoundsTimestampsForRestore() {
```

Comment:
maybe this makes sense if you're familiar with the code and semantics, but some comments are probably worth it to explain the main steps or logic that we're testing here

Reply:
added a fairly wordy comment.